srv/salt/_modules/osd.py

```diff
@@ -232,6 +233,271 @@ def tree_from_any():
     return _tree(**kwargs)
 
 
+def _get_parent_device_name(device_name):
+    """
```

Comment:
This could be done nicely with pyudev
```
context = pyudev.Context()
device = pyudev.Devices.from_device_file(context, '/dev/{}'.format(device_name))
return basename(device.parent.device_node)
```

Reply:
If this works, I'd say leave it as is. This info will be provided by ceph-volume in the very near future.

Reply:
Had to revert all pyudev changes because the library behaves different within the Salt environment and running the code manually in a Python CLI on the Ceph node.

Maybe it is somehow related to:
https://github.com/pyudev/pyudev/issues/225
https://github.com/openattic/openattic-docker/issues/9